testing/trino-product-tests/src/main/java/io/trino/tests/product/hudi/TestHudiMinioReads.java

```diff
@@ -0,0 +1,240 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.product.hudi;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.tempto.BeforeTestWithContext;
+import io.trino.tempto.ProductTest;
+import io.trino.tempto.assertions.QueryAssert;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+import static io.trino.tempto.assertions.QueryAssert.Row.row;
+import static io.trino.tempto.assertions.QueryAssert.assertThat;
+import static io.trino.tests.product.TestGroups.HUDI;
+import static io.trino.tests.product.TestGroups.PROFILE_SPECIFIC_TESTS;
+import static io.trino.tests.product.hive.util.TemporaryHiveTable.randomTableSuffix;
+import static io.trino.tests.product.utils.QueryExecutors.onHudi;
+import static io.trino.tests.product.utils.QueryExecutors.onTrino;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class TestHudiMinioReads
+        extends ProductTest
+{
+    protected String bucketName;
+
+    @BeforeTestWithContext
+    public void setUp()
+    {
+        bucketName = requireNonNull(System.getenv("S3_BUCKET"), "Environment variable not set: S3_BUCKET");
+    }
+
+    @Test(groups = {HUDI, PROFILE_SPECIFIC_TESTS})
+    public void testCopyOnWriteTableSelect()
+    {
+        String tableName = "test_hudi_demo_" + randomTableSuffix();
+        String tableDirectory = "hudi-compatibility-test-" + tableName;
+
+        createNonPartitionedHudiTable(tableName, tableDirectory, bucketName);
+
+        List<QueryAssert.Row> expectedRows = ImmutableList.of(
+                row(1, "Trino"),
+                row(2, "rocks"));
+
+        try {
+            assertThat(onHudi().executeQuery("SELECT uuid, col FROM default." + tableName))
+                    .containsOnly(expectedRows);
+            assertThat(onTrino().executeQuery("SELECT uuid, col FROM hudi.default." + tableName))
+                    .containsOnly(expectedRows);
+        }
+        finally {
+            onHudi().executeQuery("DROP TABLE default." + tableName);
+        }
+    }
+
+    @Test(groups = {HUDI, PROFILE_SPECIFIC_TESTS})
+    public void testCopyOnWritePartitionedTableSelect()
```

Comment:
Does Hudi allow specifying partitioned columns whose order is different from the table definition? e.g. CREATE TABLE (c1 int, c2 int, c3 int) PARTITIONED BY (c2, c1). Recently, Delta Lake connector faced correctness issue due to this order difference. It would be nice to have such test if Hudi supports it.

Reply:
It should be possible. I will check.